File: app/src/main/java/com/lskycity/androidtools/AppConstants.java
package com.lskycity.androidtools;

import android.content.Context;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.support.customtabs.CustomTabsIntent;
import android.support.v4.content.res.ResourcesCompat;

/**
 * Created by zhaofliu on 2/5/17.
 */

public class AppConstants {
    public static final String SHARED_KEY_INFORM_ID = "shared_key_inform_id";
    public static final String SHARED_KEY_INFORM_CONTENT = "shared_key_inform_content";
    public static final String SHARED_KEY_INFORM_CHECK_TIME = "shared_key_inform_check_time";



    public static final String INFORM_URL = "http://www.lskycity.com/androidtools/inform.json";
    public static final String CHECK_VERSION_URL = "http://www.lskycity.com/androidtools/latestversion.json";
    public static final String MAIN_PAGE_URL = "http://www.lskycity.com/androidtools/index.html";
    public static final String FEEDBACK_URL = "http://www.lskycity.com:5000/feedback";

    public static void startUrlWithCustomTab(Context context, String url) {
        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
        builder.setCloseButtonIcon(BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_arrow_back));
        builder.setToolbarColor(ResourcesCompat.getColor(context.getResources(), R.color.colorPrimary, context.getTheme()));
        builder.addDefaultShareMenuItem();
        builder.setShowTitle(true);
        CustomTabsIntent customTabsIntent = builder.build();
        customTabsIntent.launchUrl(context, Uri.parse(url));
    }
}

File: app/src/main/java/com/lskycity/androidtools/AppConstants.kt
package com.lskycity.androidtools

import android.content.Context
import android.graphics.BitmapFactory
import android.net.Uri
import androidx.browser.customtabs.CustomTabsIntent
import androidx.core.content.res.ResourcesCompat

/**
 * Created by zhaofliu on 2/5/17.
 */

object AppConstants {
    const val SHARED_KEY_INFORM_ID = "shared_key_inform_id"
    const val SHARED_KEY_INFORM_CONTENT = "shared_key_inform_content"
    const val SHARED_KEY_INFORM_CHECK_TIME = "shared_key_inform_check_time"


    const val INFORM_URL = "http://www.lskycity.com/androidtools/inform.json"
    const val CHECK_VERSION_URL = "http://www.lskycity.com/androidtools/latestversion.json"
    const val MAIN_PAGE_URL = "http://www.lskycity.com/androidtools/index.html"
    const val FEEDBACK_URL = "http://www.lskycity.com:5000/feedback"

    fun startUrlWithCustomTab(context: Context, url: String) {
        val builder = CustomTabsIntent.Builder()
        builder.setCloseButtonIcon(BitmapFactory.decodeResource(context.resources, R.drawable.ic_arrow_back))
        builder.setToolbarColor(ResourcesCompat.getColor(context.resources, R.color.colorPrimary, context.theme))
        builder.addDefaultShareMenuItem()
        builder.setShowTitle(true)
        val customTabsIntent = builder.build()
        customTabsIntent.launchUrl(context, Uri.parse(url))
    }
}
